src/modules/dashboard/components/ColonyHome/ColonyHome.tsx

```diff
@@ -220,21 +239,72 @@ const ColonyHome = ({
     />
   );
 
+  const renderNewTaskButton = (
+    <>
+      {canCreateTask && (
+        <ActionButton
+          button={({ onClick, disabled, loading }) => (
+            <Button
+              appearance={{ theme: 'primary', size: 'medium' }}
+              text={MSG.newTaskButton}
+              disabled={disabled}
+              loading={loading}
+              onClick={throttle(onClick, 2000)}
+            />
+          )}
+          disabled={isInRecoveryMode}
+          error={ActionTypes.TASK_CREATE_ERROR}
+          submit={ActionTypes.TASK_CREATE}
+          success={ActionTypes.TASK_CREATE_SUCCESS}
+          transform={transform}
+          loading={isTaskBeingCreated}
+        />
+      )}
+    </>
+  );
+
+  const crumbs =
+    domains &&
+    domains
+      .sort((a, b) => a.id - b.id)
+      .reduce((accumulator, element) => {
+        if (element && element.name && element.id <= filteredDomainId) {
+          const message = {};
+          message[element.name] = {
+            id: element.name,
+            defaultMessage: element.name,
+          };
+          const messageDescriptor = defineMessages(message);
```

Comment:
`defineMessages` is not supposed to be used in a loop (like here). Messages have to be statically defined in the head of the file (like we always do).

Reply:
Mh, then I don't know how to do it since that's specifically what Raul asked me to do, to create messageDescriptors.

Reply:
Yes you should create messageDescriptors, just not in a loop.

We have to statically define them (we know what to expect here, otherwise translations won't be possible). Maybe @rdig can also chime in?

Reply:
Could we just add new MessageDescriptors to the statically defined one?

Reply:
These ones? https://github.com/JoinColony/colonyDapp/pull/1860/files/d4d0ba5a98050a540ff96f5d0611462d85084986#diff-d800e9f821d34745e078f8a1a78854f6R49

I don't see why not!

Reply:
@chmanie Ok, I see now why not. You can not assign anything to the MSG object.

Reply:
Yeah, because it has to be _statically_ defined. What do you want to assign?

Reply:
Looking at this from another point of view:

Does the `Root` domain actually need a translation? We can just assume it's going to be called `Root` in all languages.

Reply:
Nobody actually gave a useful suggestion here. I need to dynamically create new translations, fine it's static, so I can't but what's could be a workaround then?

Reply:
The only copy that needs to be (or can be) translated is static copy that's hardcoded within the dapp. User-entered data (such as domains, in this case) cannot be translated, and shouldn't have i18n messages.

> Does the Root domain actually need a translation? We can just assume it's going to be called Root in all languages.

I think it'd be wise to translate root wherever we can. There's a default `root` message provided in the global i18n file as of #1799 being merged.

Reply:
Ok, true, so we'll use the global root message and keep all other domains as they are.  That will simplify the BreadCrumb component to the way it was like a week ago. Objections anyone?

Reply:
To me that seems like a good plan. It’s like Curtis said, thank you for the explanation. We can’t translate dynamic values, as when you think about it we have to expect beforehand which translations there _can_ be. Think about a person tasked with translating the whole app to Spanish. We need all the texts there can possibly be in English first, like in a document (just in this case it’s a JSON file). That’s what statically means in this case.